### PR TITLE
Implemented hash comparison for SCD Type 2 updates

### PR DESCRIPTION
**Pull request by polinabee**
# Description
In its current state, the SCD plugin creates new entries for all rows present in the table with a new timestamp. In cases where the timestamp is auto-generated, fields will be treated as new even if their values have not changed.
In this PR, I implement row comparison when incrementing. Each row is hashed using the user's hash algorithm of choice, and the incremental table is incremented with only those rows where the new hash does not match the hash of the currently valid entry for the same unique ID.
I am using it in my code, and figured it could be useful to others that are looking to use SCD Type 2 with a hash comparison.

# Changes in the code
- `definitions/example.js`: added the `hash` field to the variable definition and columns list
- `definitions/source_data.sqlx`: updated the example query to generate data with a hash of the `md5` algorithm
- `index.js`: modified the incremental table build to filter for rows where the hash does not match the hash in the current table for the same unique key

# Potential Improvements
Backwards compatibility: it would be good to allow the user to choose whether they want to use a hash or not, and for both the legacy method and the hash method to work with the same code.
I do not normally work with javascript so I'm sure there are possible improvements in the logic as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,9 @@ scd("source_data_scd", {
   // A field that stores a timestamp or date of when the row was last changed.
   timestamp: "updated_at",
   // The source table to build slowly changing dimensions from.
-  source: {
+    hash: "hash_value",
+    // A field that stores the hash value of the fields that we want to track changes in
+    source: {
```
Comment (lewish) [inline README.md:30]: nit: indentation off (-2)

File: index.js [hunk 2]
```diff
@@ -12,11 +12,18 @@ module.exports = (
     columns,
     ...incrementalConfig,
   }).query(
```
Comment (lewish) [inline index.js:14]: I think we should try to be backwards compatible here, and I think that's fairly simple. If the `hash` argument is missing, we just revert to the original logic, something like:
```suggestion
  }).query(
  !!hash ?
  <your new code>
  :
  <the old code>;
```

**Review by lewish — CHANGES_REQUESTED** [review]
Thanks for this! I think we should aim for backwards compatibility otherwise LGTM

**Comment by lewish** [comment]
Could I also request that you update the README to add your new hash value with a comment saying what it does, and that it is optional, similar to the other fields?

File: index.js (hunk 2 above)
Comment (polinabee) [reply index.js:14]: For sure! I'll get that added. TYSM for the quick review!

Comment (polinabee) [reply index.js:14]: Just added this and ran it both ways with test data

**Comment by polinabee** [comment]
> Could I also request that you update the README to add your new hash value with a comment saying what it does, and that it is optional, similar to the other fields?

I've added a section in the README and some additional inline comments, lmk if it's in the format you prefer. A bit wordy, but  I wanted to explain it fully

File: definitions/example.js [hunk 3]
```diff
@@ -27,5 +29,7 @@ const { updates, view } = scd("source_data_scd", {
 
 // Additional customization of the created models can be done by using the returned actions objects.
 updates.config({
+  // You can specify the output schema here if it is different than the default
+  schema: "dataform_scd_example",
```
Comment (polinabee) [inline definitions/example.js:33]: I also added this line, can be removed if you want

File: README.md [hunk 4]
```diff
@@ -25,9 +25,11 @@ scd("source_data_scd", {
   // A field that stores a timestamp or date of when the row was last changed.
   timestamp: "updated_at",
   // The source table to build slowly changing dimensions from.
-  source: {
-    schema: "dataform_scd_example",
-    name: "source_data",
+    hash: "hash_value", // OPTIONAL
+    // A field that stores the hash value of the fields that we want to track changes in. If you do not want to use the hash comparison, you may omit this field or set it to null
```
Comment (lewish) [inline README.md:27]: I think the comments in this file are above the property, not below - so you want to swap these, and move them above `// The source table to build...` line

**Review by lewish — APPROVED** [review]
Just the one comment, otherwise LGTM!

File: README.md (hunk 4 above)
Comment (polinabee) [reply README.md:27]: Right, fixed!

**Comment by polinabee** [comment]
@lewish I've addressed all the comments, let me know if there's anything else!
